src/update_roster.R

```diff
@@ -195,75 +195,81 @@ build_rosters_weekly_dataexchange <- function(season) {
+    team_week_pair <- teams |>
+      dplyr::select(team_id) |>
+      dplyr::mutate(join = 1) |>
+      dplyr::left_join(weeks |>
+                         dplyr::mutate(join = 1), by = c("join")) |>
+      dplyr::select(-c(join)) |>
+      dplyr::rename(teamId = team_id,
+                    seasonType = game_type) |>
       dplyr::mutate(
-        game_type = dplyr::case_when(
-          game_type == "POST" & week == 1 ~ "WC",
-          game_type == "POST" &
-            week == 2 ~ "DIV",
-          game_type == "POST" &
-            week == 3 ~ "CON",
-          game_type == "POST" &
-            week == 4 ~ "SB",
-          T ~ game_type
-        ),
-        week = dplyr::case_when(
-          game_type %in% c("WC", "DIV", "CON", "SB") ~ week + max(week[game_type == "REG"]),
-          T ~ week
-        ),
+        season = !!season,
+        seasonType = dplyr::case_when(seasonType != "REG" ~ "POST",
+                                      T ~ seasonType)
       )
-    weekly_rosters[["birth_date"]] <-
-      NULL # sometimes ngsscrapR::scrape_rosters() returns this, sometimes it doesn't
-  })
+    cli::cli_alert_info("Scraping rosters for {season}...")
+    scrape_rosters <- function() {
+      p <- progressr::progressor(along = 1:nrow(team_week_pair))
+      weekly_rosters <-
+        purrr::pmap_dfr(
+          list(
+            team_week_pair$teamId,
+            team_week_pair$season,
+            team_week_pair$seasonType,
+            team_week_pair$week
+          ),
+          \(w, x, y, z) {
+            roster <- ngsscrapR::scrape_roster(w, x, y, z)
+            p()
+            return(roster)
+          }
+        )
+    }
+
+    progressr::with_progress({
+      weekly_rosters <- scrape_rosters() |>
+        dplyr::mutate(years_exp = as.integer(season) - as.integer(entry_year)) |>
+        dplyr::rename(
+          position = position_group,
+          depth_chart_position = position,
+          game_type = season_type
+        ) |>
+        dplyr::group_by(game_type) |>
+        dplyr::mutate(week = dplyr::dense_rank(week)) |> # fixing some weirdness where we skip a week
+        dplyr::ungroup() |>
+        dplyr::mutate(
+          game_type = dplyr::case_when(
+            game_type == "POST" & week == 1 ~ "WC",
+            game_type == "POST" &
+              week == 2 ~ "DIV",
+            game_type == "POST" &
+              week == 3 ~ "CON",
+            game_type == "POST" &
+              week == 4 ~ "SB",
+            T ~ game_type
+          ),
+          week = dplyr::case_when(
+            game_type %in% c("WC", "DIV", "CON", "SB") ~ week + max(week[game_type == "REG"]),
+            T ~ week
+          ),
+        )
+      weekly_rosters[["birth_date"]] <-
+        NULL # sometimes ngsscrapR::scrape_rosters() returns this, sometimes it doesn't
+    })
+  } else {
+    cli::cli_alert_warning("No rosters found for {season}! It is possible the league year has not flipped over yet.")
+    weekly_rosters <- NULL
```

Comment:
Preference for early returns and avoiding else blocks if possible, eg if(length(teams)==0) return(NULL)